ambry-api/src/main/java/com/github/ambry/quota/QuotaSource.java

```diff
@@ -13,15 +13,46 @@
  */
 package com.github.ambry.quota;
 
+import com.github.ambry.messageformat.BlobInfo;
+import com.github.ambry.rest.RestRequest;
 import java.util.Collection;
 import java.util.List;
-
+import java.util.Map;
 
 /**
  * Interface representing the backend source from which quota for a resource can be fetched, and to which the current
  * usage of a resource can be saved.
  */
 public interface QuotaSource {
+
+  /**
+   * Check if {@link QuotaSource} is initialized and ready.
+   * @return {@code true} if the source is initialized and ready. {@code false} otherwise.
+   */
+  boolean isReady();
+
+  /**
+   * Charge the usage cost specified in requestCostMap against quota for resource in {@link RestRequest}.
+   * @param restRequest {@link RestRequest} for which quota should be charged.
+   * @param blobInfo {@link BlobInfo} object representing the blob for which quota is being charged.
+   * @param requestCostMap {@link Map} of {@link QuotaName} to cost incurred for serving the request.
+   */
+  void charge(RestRequest restRequest, BlobInfo blobInfo, Map<QuotaName, Double> requestCostMap);
+
+  /**
+   * Get {@link QuotaRecommendation} based on usage of resource specified in the {@link RestRequest}.
+   * @param restRequest {@link RestRequest} object.
+   * @return QuotaRecommendation after checking usage.
+   */
+  QuotaRecommendation checkResourceUsage(RestRequest restRequest);
+
+  /**
+   * Get {@link QuotaRecommendation} based on frontend usage.
+   * @param restRequest {@link RestRequest} object.
+   * @return QuotaRecommendation after checking usage.
+   */
+  QuotaRecommendation checkFrontendUsage(RestRequest restRequest);
```

Comment:
why are we exposing the frontend usage in the quota source as opposed to hide it in the capacity unit enforcer?